controls/V-72027.rb

```diff
@@ -1,40 +1,34 @@
 # encoding: utf-8
 #
-=begin
------------------
-Benchmark: Red Hat Enterprise Linux 7 Security Technical Implementation Guide
-Status: Accepted
 
-This Security Technical Implementation Guide is published as a tool to improve
-the security of Department of Defense (DoD) information systems. The
-requirements are derived from the National Institute of Standards and
-Technology (NIST) 800-53 and related documents. Comments or proposed revisions
-to this document should be sent via email to the following address:
-disa.stig_spt@mail.mil.
+EXEMPT_HOME_USERS = attribute(
+  'exempt_home_users',
+  description: 'These are `home dir` exempt interactive accounts',
+  default: []
+)
 
-Release Date: 2017-03-08
-Version: 1
-Publisher: DISA
-Source: STIG.DOD.MIL
-uri: http://iase.disa.mil
------------------
-=end
+NON_INTERACTIVE_SHELLS = attribute(
```

Comment:
This should be pulled from the system and reversed.

The login shells are those that are listed in `/etc/shells`. No other shells will be respected for user logins.

Reply:
The `/etc/shells` does not seem to do enforcement at login time. I changed my shell to `zsh` just by editing `/etc/passwd` even though it's not listed in my `/etc/shell`, and it allowed me to login just fine. The purpose of the list is to exclude users who are using "shells" which we know are not interactive in order to flag them as non-interactive users. It's not to validate that a shell should be allowed.

Reply:
@samcornwell If you use the `chsh` command it should not let you do that. Yes, hand editing `/etc/passwd` may work in some cases but that should generally be flagged as VERY suspicious behavior.

Reply:
@trevor-vaughan Thats fair, but it doesn't really have anything to do with this control anyway. As I said, the purpose of the list is to exclude users who are using "shells" which we know are not interactive in order to flag them as non-interactive users. It seems the default `/etc/shells` by default on CentOS 7 (at least on the bento vagrant box) includes the `nologin` shell, which means that it will not be a reliable way of doing what we're trying to do here. It may be a bad idea to have `nologin` in `/etc/shells`, but it's an unrelated issue which can cause these controls to fail even though they have nothing to do with that.